src/libNode/MicroBlockProcessing.cpp

```diff
@@ -195,11 +195,58 @@ bool Node::ProcessMicroblockConsensus(const vector<unsigned char>& message,
         LOG_EPOCH(WARNING, to_string(m_mediator.m_currentEpochNum).c_str(),
                   "Oops, no consensus reached - what to do now???");
 
+        if (m_consensusObject->GetConsensusErrorCode()
+            == ConsensusCommon::MISSING_TXN)
+        {
+            // Missing txns in microblock proposed by leader. Will attempt to fetch
+            // missing txns from leader, set to a valid state to accept cosig1 and cosig2
+            LOG_EPOCH(WARNING, to_string(m_mediator.m_currentEpochNum).c_str(),
+                      "Oops, no consensus reached - consensus error. "
+                      "error number: "
+                          << (m_consensusObject->GetConsensusErrorCode())
+                          << " error message: "
+                          << (m_consensusObject->GetConsensusErrorMsg()));
+
+            // Block till txn is fetched
+            unique_lock<mutex> lock(m_mutexCVMicroBlockMissingTxn);
+            if (cv_MicroBlockMissingTxn.wait_for(
```

Comment:
Should notify be called on cv_MicroBlockMissingTxn somewhere?

Reply:
yep. I also notice a notify is missing